### PR TITLE
Adding documentation to details component

### PR DESCRIPTION
**Pull request by mikegiann01**

### CODE REVIEW

**Comment by mikegiann01** [comment]
This is about issue #222

File: solara/components/details.py [hunk 1]
```diff
@@ -19,3 +19,36 @@ def on_v_model(v_model):
...
+
+
+    ## Arguments
+
+    *summary: A string representing the summary text for the expandable section. It defaults to "Summary".
+    *children: A list representing the children content of the expandable section. It defaults to an empty list.
+    *expand: A boolean representing whether the section is initially expanded or collapsed. It defaults to False.
+
+
+    """
```
Comment (iisakkirotko) [inline solara/components/details.py:54]: I think these should be moved to be right under the `def Details` Line

**Review by iisakkirotko — CHANGES_REQUESTED** [review]
Hi @mikegiann01!

Thanks for the PR! The code quality errors should be easy to fix. Could you run 
`pip install pre-commit`
followed by
`pre-commit install`
This should automatically apply the formatting that pre-commit is complaining about when you try to run `git commit`. 

I also notice that `solara.Details` isn't present in the documentation right now. Would you like to also create a page for it in there? You can take a look at [`solara/website/pages/documentation/components/layout/card.py`](https://github.com/widgetti/solara/blob/master/solara/website/pages/documentation/components/layout/card.py) to see how it's done. Let me know here or on Discord if you have any questions!

**Comment by mikegiann01** [comment]
Thanks for the response! I will follow your istructions and i will let you know on the process. About the documentation on the website i am very interested on working on this, i will check it out and i will let you know.

**Comment by mikegiann01** [comment]
Hello!! I have some upadates, I followed your instructions and i reached an impasse(shown in the picture):
error
Could you maybe pinpoint what im doing wrong?

**Comment by iisakkirotko** [comment]
Hey @mikegiann01!

I rebased the branch, so you could pull and try committing again. If that doesn't work, try upgrading flake8, or uninstalling and reinstalling pre-commit. (`pre-commit clean`, or `pre-commit uninstall` and `pre-commit install`)

**Review by iisakkirotko — CHANGES_REQUESTED** [review]
Hi @mikegiann01!

Looks like you're still having issues with pre-commit. It seems like the config file (`.pre-commit-config.yaml`) in your fork has been reset to a state from before some recent changes, which means that the code is being judged against different guidelines on your machine than online in our repository. I think this should be fixed with the following stes:
- reset `.pre-commit-config.yaml` to the state it is in the master branch of our repo
- run `pre-commit uninstall && pre-commit install` to install pre-commit

Other changes I'd like to see before merging this:
- remove the newly added pre-commit config files (`solara/.pre-commit-config.yaml`, etc.)
- squash the current 6 commits down to one or two (using, for example, `git rebase -i HEAD~6`)

**Comment by mikegiann01** [comment]
Okay thank you for the response!! Sorry for the number of commits and the
way i ve been using the platform , I am still trying to learn.

Στις Πέμ 18 Απρ 2024 στις 10:06 π.μ., ο/η Iisakki Rotko <
***@***.***> έγραψε:

**Pull request closed without merging**